src/Rules/Documentation.hs

```diff
@@ -10,6 +10,81 @@ import Settings
 import Target
 import Utilities
 
+docPackage :: Package
+docPackage = hsLibrary "Documentation" "docs"
+
+-- | Build all documentation
+documentationRules :: Rules ()
+documentationRules = do
+    buildHtmlDocumentation
+    buildPdfDocumentation
+    buildDocumentationArchives
+    "docs" ~> do
+        root <- buildRoot
+        -- Archives `need` the associated html
+        let archives = map pathArchive docPaths
+            pdfs = map pathPdf $ drop 1 docPaths
+        need $ map (root -/-) $ archives ++ pdfs
+
+docPaths :: [FilePath]
+docPaths = [ "libraries", "users_guide", "Haddock" ]
+
+docRoot :: FilePath
+docRoot = "docs"
+
+htmlRoot :: FilePath
+htmlRoot = docRoot -/- "html"
+
+pdfRoot :: FilePath
+pdfRoot = docRoot -/- "pdfs"
+
+archiveRoot :: FilePath
+archiveRoot = docRoot -/- "archives"
+
+pathPdf :: FilePath -> FilePath
+pathPdf path = pdfRoot -/- path <.> ".pdf"
+
+pathIndex :: FilePath -> FilePath
+pathIndex path = htmlRoot -/- path -/- "index.html"
+
+pathLibIndex :: FilePath -> FilePath
+pathLibIndex path = htmlRoot -/- "libraries" -/- path -/- "index.html"
+
+pathArchive :: FilePath -> FilePath
+pathArchive path = archiveRoot -/- path <.> "html.tar.xz"
+
+pathPath :: FilePath -> FilePath
+pathPath "users_guide" = "docs/users_guide"
+pathPath "Haddock" = "utils/haddock/doc"
+pathPath _ = "lol"
```

Comment:
Looks like we might want a data type here, e.g. `data Doc = UserGuide | Haddock` or something similar?

Reply:
Actually, looking at how `pathPath` is used, my suggestion doesn't work. If we leave this as is, we better add a `TODO` to get rid of this hack later.